netket/graph/graph.py

```diff
@@ -12,184 +12,171 @@
-        constructed specifying the edges as follows:
-
-        >>> import netket
-        >>> g=netket.graph.Graph(edges=[[i, (i + 1) % 10] for i in range(10)])
-        >>> print(g.n_nodes)
-        10
-
-    """
-    if not isinstance(nodes, list):
-        raise TypeError("nodes must be a list")
-
-    if not isinstance(edges, list):
-        raise TypeError("edges must be a list")
-
-    if edges:
-        type_condition = [
-            isinstance(edge, list) or isinstance(edge, tuple) for edge in edges
-        ]
-        if False in type_condition:
-            raise ValueError("edges must be a list of lists or tuples")
-
-        edges_array = np.array(edges, dtype=np.int32)
-        if edges_array.ndim != 2:
-            raise ValueError(
-                "edges must be a list of lists or tuples of the same length (2 or 3)"
-            )
+    # Graph algorithms
+    # ------------------------------------------------------------------------
+    def distances(self) -> List[List]:
+        return np.array(self._igraph.shortest_paths())
 
-        if not (edges_array.shape[1] == 2 or edges_array.shape[1] == 3):
-            raise ValueError(
-                "edges must be a list of lists or tuples of the same length (2 or 3), where the third column indicates the color"
-            )
+    def _compute_automorphisms(self):
+        """
+        Compute the graph autmorphisms of this graph.
+        """
+        colors = self._edge_colors
+        result = self._igraph.get_isomorphisms_vf2(
+            edge_color1=colors, edge_color2=colors
+        )
 
-        # Sort node names for ordering reasons:
-    if nodes:
-        node_names = sorted(nodes)
-    elif edges:
-        node_names = sorted(set((node for edge in edges_array for node in edge)))
+        # sort them s.t. the identity comes first
+        result = np.unique(result, axis=0).tolist()
+        result = PermutationGroup([Permutation(i) for i in result], self.n_nodes)
+        return result
 
-    graph = _nx.MultiGraph()
-    graph.add_nodes_from(node_names)
-    if edges:
-        graph.add_edges_from(edges_array)
-        if edges_array.shape[1] == 3:  # edges with color
-            colors = {tuple(e): e[-1] for e in edges}
-            _nx.set_edge_attributes(graph, colors, name="color")
-        else:  # only one color
-            _nx.set_edge_attributes(graph, 0, name="color")
+    # TODO turn into a struct.property_cached?
+    def automorphisms(self) -> PermutationGroup:
+        if self._automorphisms is None:
+            self._automorphisms = self._compute_automorphisms()
+        return self._automorphisms
 
-    return NetworkX(graph)
+    # Output and drawing
+    # ------------------------------------------------------------------------
+    def __repr__(self):
+        return "{}(n_nodes={}, n_edges={})".format(
+            str(type(self)).split(".")[-1][:-2], self.n_nodes, self.n_edges
+        )
 
```

Comment:
a fancy thing we might do in a future refinement is to pretty-print this automatically in Jupiter notebooks by plotting the graph itself.
It should be almost a 2 liner

Reply:
We could also just add `.draw()` maybe, like for lattice? Anyways, I'd do that in a separate PR. It's a nice idea though.